src/HallPIC.jl

```diff
@@ -630,22 +630,47 @@ struct Reaction{T<:AbstractFloat}
     threshold_energy::T
     rate_table::LinearInterpolation
     delta_n::Vector{T}
+    delta_n_remainder::Vector{T}
 
 end
 
 
 
-function deplete_reactant!(reaction::Reaction{T}, reactant::ParticleContainer{T}, fluid_properties::SpeciesProperties{T}, grid::Grid, electron_properties::SpeciesProperties{T}, dt) where T
+function deplete_reactant!(reaction::Reaction{T}, reactant::ParticleContainer{T}, reactant_properties::SpeciesProperties{T}, products::Vector{ParticleContainer{T}}, product_properties::Vector{SpeciesProperties{T}}, grid::Grid, electron_properties::SpeciesProperties{T}, dt) where T
 
     # for each (non-ghost) cell, calculate the delta n
```

Comment:
comment: delta n -> loss in density due to the reaction or something more explanatory